src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/FindCommandParserTest.java

```diff
@@ -1,40 +1,266 @@
+        predicateList.add(new InDateRangePredicate(lowerBound, Optional.empty()));
+        FindCommand expectedFindCommand = new FindCommand(predicateList);
+        assertParseSuccess(parser, DATE_FROM_DESC_INTERNSHIP, expectedFindCommand);
+
+        // upper bound only
+        predicateList = new ArrayList<>();
+        predicateList.add(new InDateRangePredicate(Optional.empty(), upperBound));
+        expectedFindCommand = new FindCommand(predicateList);
+        assertParseSuccess(parser, DATE_TO_DESC_BUBBLE_TEA, expectedFindCommand);
+
+        // upper and lower bound
+        predicateList = new ArrayList<>();
+        predicateList.add(new InDateRangePredicate(lowerBound, upperBound));
+        expectedFindCommand = new FindCommand(predicateList);
+        assertParseSuccess(parser, DATE_FROM_DESC_INTERNSHIP + DATE_TO_DESC_BUBBLE_TEA, expectedFindCommand);
+
+        // same upper and lower bound values
+        predicateList = new ArrayList<>();
+        predicateList.add(new InDateRangePredicate(upperBound, upperBound));
+        expectedFindCommand = new FindCommand(predicateList);
+        assertParseSuccess(parser, DATE_FROM_DESC_BUBBLE_TEA + DATE_TO_DESC_BUBBLE_TEA, expectedFindCommand);
+    }
+
+    @Test
+    public void parse_multipleSearchParameters_success() {
+        List<Predicate<Transaction>> predicateList = new ArrayList<>();
+        predicateList.add(new TitleContainsKeyphrasesPredicate(Arrays.asList(VALID_TITLE_BUBBLE_TEA)));
+        predicateList.add(new HasExactAmountPredicate(new Amount(VALID_AMOUNT_BUBBLE_TEA)));
+        FindCommand expectedFindCommand = new FindCommand(predicateList);
+        assertParseSuccess(parser, TITLE_DESC_BUBBLE_TEA + AMOUNT_DESC_BUBBLE_TEA, expectedFindCommand);
+    }
+
+    @Test
+    public void parse_invalidCommandFormat_failure() {
+        assertParseFailure(parser, PREAMBLE_WHITESPACE,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, PREAMBLE_NON_EMPTY,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_hasEmptyArgs_failure() {
+        // empty title keyphrase
+        assertParseFailure(parser, " " + PREFIX_TITLE, MESSAGE_EMPTY_TITLE_KEYPHRASE);
+
+        // empty title keyphrase followed by non-empty title keyphrase
+        assertParseFailure(parser, " " + PREFIX_TITLE + TITLE_DESC_BUBBLE_TEA, MESSAGE_EMPTY_TITLE_KEYPHRASE);
+
+        // empty category
+        assertParseFailure(parser, " " + PREFIX_CATEGORY, MESSAGE_EMPTY_CATEGORY);
+
+        // empty category followed by non-empty category
+        assertParseFailure(parser, " " + PREFIX_CATEGORY + CATEGORY_DESC_FOOD_BEVERAGE, MESSAGE_EMPTY_CATEGORY);
+
+        // multiple empty search parameters, first error message thrown
+        assertParseFailure(parser, " " + PREFIX_TITLE + " " + PREFIX_CATEGORY, MESSAGE_EMPTY_TITLE_KEYPHRASE);
```

Comment:
Any reason for prepending with a space here?

Reply:
The spaces are necessary in order for the parsing to work.